src/Symfony/Component/DependencyInjection/Context.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Exception\BadMethodCallException;
+use Symfony\Component\DependencyInjection\Exception\ContextElementNotFoundException;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+final class Context
+{
+    private $elements;
+    private $locked = false;
+
+    public function __construct(array $context = array())
+    {
+        $this->elements = $context;
+    }
+
+    public function has($key)
+    {
+        return isset($this->elements[$key]);
+    }
+
+    public function get($key)
+    {
+        if (!$this->has($key)) {
+            throw new ContextElementNotFoundException($key);
+        }
+
+        return $this->elements[$key];
+    }
+
+    public function set($key, $element)
+    {
+        if ($this->isLocked()) {
+            throw new BadMethodCallException(sprintf('Setting element "%s" on a locked context is not allowed.', $key));
+        }
+
+        $this->elements[$key] = $element;
+    }
+
+    public function lock()
+    {
+        $this->locked = true;
+    }
+
+    public function isLocked()
```

Comment:
Missing phpdoc block to add `@return bool`? Granted its obvious looking at the code, but as we don't have the typehint in the method signature